### PR TITLE
[Merged by Bors] - chore: avoid importing `Cardinal` in basic group theory

### PR DESCRIPTION
**Pull request by kim-em**

### CODE REVIEW

File: Mathlib/GroupTheory/Commutator/Basic.lean [hunk 1]
```diff
@@ -229,3 +209,6 @@ theorem mem_commutatorSet_iff : g ∈ commutatorSet G ↔ ∃ g₁ g₂ : G, ⁅
 
 theorem commutator_mem_commutatorSet : ⁅g₁, g₂⁆ ∈ commutatorSet G :=
   ⟨g₁, g₂, rfl⟩
+
+assert_not_exists Cardinal
+assert_not_exists Multiset
```
Comment (Ruben-VandeVelde) [inline Mathlib/GroupTheory/Commutator/Basic.lean:214]: Can you please put those at the top?

File: Mathlib/GroupTheory/Coset/Basic.lean [hunk 2]
```diff
@@ -722,3 +686,5 @@ noncomputable def preimageMkEquivSubgroupProdSet (s : Subgroup α) (t : Set (α
   right_inv := fun ⟨⟨a, ha⟩, ⟨x, hx⟩⟩ => by ext <;> simp [ha]
 
 end QuotientGroup
+
+assert_not_exists Cardinal
```
Comment (Ruben-VandeVelde) [inline Mathlib/GroupTheory/Coset/Basic.lean:690]: Ditto

File: Mathlib/Topology/Algebra/Ring/Basic.lean [hunk 3]
```diff
@@ -354,3 +354,5 @@ def AbsoluteValue.comp {R S T : Type*} [Semiring T] [Semiring R] [OrderedSemirin
   add_le' _ _ := (congr_arg v (map_add f _ _)).trans_le (v.add_le _ _)
 
 end AbsoluteValue
+
+assert_not_exists Cardinal
```
Comment (Ruben-VandeVelde) [inline Mathlib/Topology/Algebra/Ring/Basic.lean:358]: Ditto

**Review by Ruben-VandeVelde — APPROVED** [review]
lgtm, though I'd prefer you keep the import statements sorted

**Comment by semorrison** [comment]
> lgtm, though I'd prefer you keep the import statements sorted

I think this is a problem with `shake`, actually. When splitting, I try to leave imports "as is" (i.e. I replace the split file with the "last" of its split children), and then just let `shake` deal with minimising imports.

My preference is not to add the busy work for humans to resort imports, and either automate resorting imports periodically, or change `shake`.

File: Mathlib/GroupTheory/Commutator/Basic.lean (hunk 1 above)
Comment (semorrison) [reply Mathlib/GroupTheory/Commutator/Basic.lean:214]: We're actually already split in Mathlib as to whether these go at the top or bottom. I'm unsure myself which is better. At the top means they are closer to the import statements, at the bottom they are less clutter in the way of a normal reader who just wants to start reading declarations.

Comment (semorrison) [reply Mathlib/GroupTheory/Commutator/Basic.lean:214]: Looking for consensus at https://leanprover.zulipchat.com/#narrow/stream/345428-mathlib-reviewers/topic/.60assert_not_exists.60.20at.20top.20or.20bottom.20of.20file.3F/near/459960523. I'll mark these as resolved in the meantime and we can cleanup later as required.

**Comment by tb65536** [comment]
maintainer merge

**Comment by mattrobball** [comment]
Yes. 

bors merge

**Pull request closed without merging**